fieldset_test.go

```diff
@@ -0,0 +1,53 @@
+package ucfg
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCanAddAField(t *testing.T) {
+	fs := makeFieldSet(nil)
+	fs.Add("hello")
+	assert.True(t, fs.Has("hello"))
+}
+
+func TestCanAFieldThatDoesntExist(t *testing.T) {
+	fs := makeFieldSet(nil)
+	assert.True(t, fs.AddNew("hello"))
+}
+
+func TestCannotAddAFieldThatExist(t *testing.T) {
+	fs := makeFieldSet(nil)
+	fs.Add("hello")
+	assert.False(t, fs.AddNew("hello"))
+}
+
+func TestReturnsTheListOfFields(t *testing.T) {
+	fs1 := makeFieldSet(nil)
+	fs1.Add("hello")
+	fs1.Add("bye")
+	fs2 := makeFieldSet(fs1)
+	fs2.Add("adios")
+	assert.ElementsMatch(t, []string{"hello", "bye", "adios"}, fs2.Names())
+}
+
+func TestRecursiveHas(t *testing.T) {
+	fs1 := makeFieldSet(nil)
+	fs1.Add("hello")
+
+	fs2 := makeFieldSet(fs1)
+
+	assert.True(t, fs2.Has("hello"))
+	assert.False(t, fs2.Has("bye"))
+}
+
+func TestRecurseAddNew(t *testing.T) {
+	fs1 := makeFieldSet(nil)
+	fs1.Add("hello")
+
+	fs2 := makeFieldSet(fs1)
+
+	assert.False(t, fs2.AddNew("hello"))
+	assert.True(t, fs2.AddNew("bye"))
+}
```

Comment:
these tests are fieldset related. Using common prefix or `t.Run` to create some grouping. The `go test -run ...` takes a regular expression. Sub-tests can be separated by `/` in the run argument. e.g. `go test -run 'TestFieldSet/has'.